src/middlewares/checkAuth.js

```diff
@@ -2,7 +2,7 @@ import localStorage from 'localStorage';
 import decodeToken from '../helpers/decodeToken';
 
 const auth = (req, res, next) => {
-  const token = localStorage.getItem('token');
+  const token = req.header('user-token') || localStorage.getItem('token');
```

Comment:
As I can see that will cause a lot of problems when a client didn't provide a token. it will use a token that is in the local storage even if he/she didn't log in or sign up. better use token from a header only.